airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py

```diff
@@ -3,30 +3,39 @@
 #
 
 import json
-from typing import Any, Mapping, Optional
+from dataclasses import InitVar, dataclass
+from typing import Any, Mapping, Union
 
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.schema.schema_loader import SchemaLoader
 from airbyte_cdk.sources.declarative.types import Config
+from dataclasses_jsonschema import JsonSchemaMixin
 
 
-class JsonSchema(SchemaLoader):
-    """Loads the schema from a json file"""
+@dataclass
+class JsonSchema(SchemaLoader, JsonSchemaMixin):
+    """
+    Loads the schema from a json file
 
-    def __init__(self, file_path: InterpolatedString, config: Config, **options: Optional[Mapping[str, Any]]):
-        """
-        :param file_path: The path to the json file describing the schema
-        :param config: The user-provided configuration as specified by the source's spec
-        :param options: Additional arguments to pass to the string interpolation if needed
-        """
-        self._file_path = file_path
-        self._config = config
-        self._options = options
+    Attributes:
+        file_path (Union[InterpolatedString, str]): The path to the json file describing the schema
+        name (str): The stream's name
+        config (Config): The user-provided configuration as specified by the source's spec
+        options (Mapping[str, Any]): Additional arguments to pass to the string interpolation if needed
+    """
+
+    file_path: Union[InterpolatedString, str]
+    name: str
+    config: Config
+    options: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, options: Mapping[str, Any]):
+        self.file_path = InterpolatedString.create(self.file_path, options=options)
 
     def get_json_schema(self) -> Mapping[str, Any]:
         json_schema_path = self._get_json_filepath()
         with open(json_schema_path, "r") as f:
             return json.loads(f.read())
 
     def _get_json_filepath(self):
-        return self._file_path.eval(self._config, **self._options)
+        return self.file_path.eval(self.config)
```

Comment:
This is actually more of a bug fix. If we pass in `self.config` and `self._options` here, we might have a name collision since config can be populated in the incoming parent options. Rather, we should pass options in during instantiation like we do other classes.